cirq/google/decompositions_test.py

```diff
@@ -11,32 +11,27 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+from typing import Iterable, List
 
 import random
 
 import numpy as np
-
 import pytest
 
 import cirq
 from cirq.google import decompositions
 
 
-def _operations_to_matrix(operations, qubits):
+def _operations_to_matrix(operations: Iterable[cirq.Operation],
+                          qubits: Iterable[cirq.QubitId]):
     return cirq.Circuit.from_ops(operations).to_unitary_matrix(
         qubit_order=cirq.QubitOrder.explicit(qubits),
         qubits_that_should_be_present=qubits)
 
 
-def _gates_to_matrix(gates):
-    m = gates[0].matrix()
-    for gate in gates[1:]:
-        m = gate.matrix().dot(m)
-    return m
-
-
-def assert_gates_implement_unitary(gates, intended_effect):
-    actual_effect = _gates_to_matrix(gates)
+def assert_gates_implement_unitary(gates: List[cirq.SingleQubitGate],
+                                   intended_effect: np.ndarray):
+    actual_effect = cirq.dot(*reversed([cirq.unitary(g) for g in gates]))
```

Comment:
I would suggest doing:
```python
actual_effect = cirq.dot(*[cirq.unitary(g) for g in reversed(gates)])
```

Reply:
Done. (You know this makes it not work on all iterables, right? But not relevant here.)